### PR TITLE
Fix Condition ' name == "" ' is true when new name contains only unallowed characters

### PR DESCRIPTION
**Pull request by ghost**
_Bugsquad edit : Fix #25226_

### CODE REVIEW

**Comment by aayushchhabra** [comment]
#25226

File: editor/scene_tree_editor.cpp [hunk 1]
```diff
@@ -673,6 +673,7 @@ void SceneTreeEditor::_renamed() {
 	}
 
 	String new_name = which->get_text(0);
+        String pristine_new_name = new_name;
```
Comment (akien-mga) [inline editor/scene_tree_editor.cpp]: You should use tabs for indentation, not spaces.

File: scene/main/node.cpp [hunk 2]
```diff
@@ -925,10 +925,9 @@ bool Node::_validate_node_name(String &p_name) {
 void Node::set_name(const String &p_name) {
 
 	String name = p_name;
-	_validate_node_name(name);
-
+        ERR_FAIL_COND(!_validate_node_name(name));
```
Comment (akien-mga) [inline scene/main/node.cpp]: That's not the correct fix, `_validate_node_name` is meant to fix the name given as input, not error out.

Comment (akien-mga) [reply scene/main/node.cpp]: Well that's up for discussion, but the current logic is that if you pass an invalid node name, this method will fix it for you and use the fixed name.

**Comment by akien-mga** [comment]
I think something like this would be better:
```diff
diff --git a/editor/scene_tree_editor.cpp b/editor/scene_tree_editor.cpp
index 7e4baa153..7e20baa9b 100644
--- a/editor/scene_tree_editor.cpp
+++ b/editor/scene_tree_editor.cpp
@@ -681,7 +681,7 @@ void SceneTreeEditor::_renamed() {
                which->set_text(0, new_name);
        }
 
-       if (new_name == n->get_name())
+       if (new_name == n->get_name() || new_name.strip_edges() == String())
                return;
 
        if (!undo_redo) {
```

File: scene/main/node.cpp (hunk 2 above)
Comment (aayushchhabra) [reply scene/main/node.cpp]: The method name is really misleading ...

File: editor/scene_tree_editor.cpp (hunk 1 above)
Comment (aayushchhabra) [reply editor/scene_tree_editor.cpp]: Will do.

**Comment by aaronfranke** [comment]
@aayushchhabra If you need help with Git, please see Godot's article on [pull request workflow](https://godot.readthedocs.io/en/latest/community/contributing/pr_workflow.html).

**Comment by aayushchhabra** [comment]
Thanks @aaronfranke but I was caught up with something else, I'll work on it today.

**Comment by akien-mga** [comment]
Superseded by #25469.

**Pull request closed without merging**